SCons/Tool/msvc.py

```diff
@@ -212,6 +213,25 @@ def msvc_output_flag(target, source, env, for_signature):
                                   batch_key=msvc_batch_key,
                                   targets='$CHANGED_TARGETS')
 
+def CCPCHFLAGS_gen(target, source, env, for_signature):
+    pch =  env.get("PCH")
+    pch_subst = env.subst("$PCH", target=target, source=source)
+
+    # we want to know if PCH was set, but we also want to know
+    # if the user supplied a generator, if the subst returned something
+    # otherwise we assume the user doesn't want PCH for this invokation
+    if pch and pch_subst:
+        
+        # if the current value odes not resolve to anything, we assume
+        # the file should be resolved to the current sconscript dir, so
+        # we subst again converting it to a local File.
+        if not env.File(pch_subst).exists():
+            pch_subst = env.subst("${File(PCH)}", target=target, source=source)
+        
+        return '/Yu%s "/Fp%s"' % ( 
+            env.subst('$PCHSTOP', target=target, source=source) or "",
```

Comment:
I did not update this PCHSTOP to support subst in other places so I should revert this to direct access.

Reply:
done: 5eb908a